Firmware/src/modules/tools/temperaturecontrol/max31855.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+      This file is part of Smoothie (http://smoothieware.org/). The motion control part is heavily based on Grbl (https://github.com/simen/grbl).
+      Smoothie is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+      Smoothie is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+      You should have received a copy of the GNU General Public License along with Smoothie. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <math.h>
+#include <fcntl.h>
+
+#include "max31855.h"
+#include "OutputStream.h"
+#include "main.h"
+
+#define designator_key           "designator"
+#define tool_id_key                 "tool_id"
+#define dev_id_key                   "dev_id"
+
+Max31855 *Max31855::instances[Max31855::num_dev] = {nullptr};
+RingBuffer<float,16> *Max31855::queue[Max31855::num_dev] = {nullptr};
+int Max31855::ninstances = 0;
+bool Max31855::thread_flag = true;
+
+/* Initialize target instance */
+Max31855::Max31855()
+{
+    this->instance_idx = ninstances++;
+    instances[this->instance_idx] = this;
+    queue[this->instance_idx] = new RingBuffer<float,16>(); // buffer initialized with 16 float elements
+    this->error_flag = false;
+}
+
+/* Deinitialize target instance */
+Max31855::~Max31855()
+{
+    // remove target instance from instances array
+    instances[this->instance_idx] = nullptr;
+    for (int i = this->instance_idx; i < ninstances - 1; ++i) {
+        instances[i] = instances[i + 1];
+        queue[i] = queue[i + 1];
+    }
+    --ninstances;
+    this->error_flag=false;
+}
+
+/* Configure the module using the parameters from the config file */
+bool Max31855::configure(ConfigReader& cr, ConfigReader::section_map_t& m)
+{
+    /* open file in Nuttx corresponding to the selected device ID:
+        dev_id 0: File /dev/temp0, Channel SSP0, Device 0
+        dev_id 1: File /dev/temp1, Channel SSP0, Device 1
+        dev_id 2: File /dev/temp2, Channel SSP1, Device 0
+        dev_id 3: File /dev/temp3, Channel SSP1, Device 1   */
+    std::string devpath = "/dev/temp";
+    this->dev_id = cr.get_int(m, dev_id_key, 0);
+    devpath += std::to_string(this->dev_id);
+    this->fd = open(devpath.c_str(), O_RDONLY);
+    // negative handler if can't successfully open, positive if it is open
+    if (this->fd < 0) {
+        printf("ERROR: Could not open file %s\n",devpath);
+        return false;
+    }
+
+    // identifying tool
+    this->designator = cr.get_string(m, designator_key, "");
```

Comment:
This is wrong tool_id and designator are set in Temperature control... NOT here.
https://github.com/Smoothieware/smoothie-v2/blob/master/Firmware/src/modules/tools/temperaturecontrol/TemperatureControl.cpp#L95